sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala

```diff
@@ -315,7 +315,12 @@ object InMemoryFileIndex extends Logging {
         // which is very slow on some file system (RawLocalFileSystem, which is launch a
         // subprocess and parse the stdout).
         try {
-          val locations = fs.getFileBlockLocations(f, 0, f.getLen)
+          val locations = fs.getFileBlockLocations(f, 0, f.getLen).map(
+            loc => if (loc.getClass == classOf[BlockLocation]) {
```

Comment:
`lo.isInstanceOf[BlockLocation]`? Or even better, what about using pattern matching?

Reply:
Thanks @mgaido91, but loc is always an instance of `BlockLocation` (might be a subclass such as `HdfsBlockLocation`) so isInstanceOf[BlockLocation] or pattern matching would return always true.
I want to test that the class of loc is exactly `BlockLocation` and if it is we don't need to convert it.

Reply:
ah right, sorry @peter-toth. Thanks. Anyway, please move `loc` to the previous line and use curly braces for map. I think that is the most widely spread syntax in the codebase. Thanks.